firefly/app.py

```diff
@@ -90,3 +98,19 @@ def make_response(self, result, status=200):
         response.text = json_encode(result)
         response.status = status
         return response
+
+    def generate_signature(self, f):
+        func_sig = signature(f)
+        params = []
+
+        for param_name, param_obj in func_sig.parameters.items():
+            param = {
+                "name": param_name,
+                "kind": str(param_obj.kind)
+            }
+            default = param_obj.default if not param_obj.default == _empty else None
+            if default:
+                param["default"] = default
```

Comment:
`is` checks for sameness where as `==` checks for equality. 
```
>>> (1, 2) == (1, 2)
True
>>> (1, 2) is (1, 2)
False
```

We are using two different tuples. They are equal, but not the same. 

When using sentinel value like `_empty`, sameness check is more appropriate.